### DISCUSSION TITLE
Feature Request: NaN -Infinity Infinity support for Decimal/Numeric types

### DISCUSSION THREAD

**Discussion by daniar-achakeev** [duckdb/duckdb#12042] [Ideas]
NaN and infinity values are useful when working with arbitrary precision numbers and financial data.
Similar to [postgres](https://www.postgresql.org/docs/current/datatype-numeric.html) decimal/numeric types.

**Comment by soerenwolfers** [comment]
Duckdb doesn't have the arbitrary precision DECIMAL type afaik

**Reply by soerenwolfers** [reply]
In particular, `DECIMAL` defaults to width 18 and scale 3

**Reply by daniar-achakeev** [reply]
Thanks! Although they are not of arbitrary precision, it is still useful to have these special values while working with financial data.

**Reply by soerenwolfers** [reply]
What I mean is: PostgreSQL offers the special values only for columns of type "unconstrained DECIMAL" and they seem to believe there is a good reason for that limitation ("Note that an infinity can only be stored in an unconstrained numeric column, because it notionally exceeds any finite precision limit.") so rather than duckdb creating something that PostgreSQL doesn't have, it's probably better to copy what PostgreSQL does entirely, including the data type itself.

**Reply by daniar-achakeev** [reply]
I agree, you are right about '-Infinity' and 'Infinity' ( 'NaN' value on contrary is supported for both decimal types (precise and unconstrained), 'NaN' is currently most useful for me :-) ).
